src/sentry/models/relocation.py

```diff
@@ -0,0 +1,252 @@
+from __future__ import annotations
+
+import logging
+from enum import Enum, unique
+from uuid import uuid4
+
+from django.db import models
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import BoundedBigIntegerField, region_silo_only_model
+from sentry.db.models.base import DefaultFieldsModel, sane_repr
+from sentry.db.models.fields.foreignkey import FlexibleForeignKey
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+from sentry.db.models.fields.uuid import UUIDField
+
+logger = logging.getLogger(__name__)
+
+
+def default_guid():
+    return uuid4().hex
+
+
+@region_silo_only_model
+class Relocation(DefaultFieldsModel):
+    """
+    Represents a single relocation instance. The relocation may be attempted multiple times, but we
+    keep a mapping of 1 `Relocation` model per file upload.
+    """
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    # The last stage this relocation reached. If the `Status` is `IN_PROGRESS`, the relocation is
+    # still active; otherwise, this can be considered the terminal stage.
+    class Step(Enum):
+        UNKNOWN = 0
+        # The upload of the underlying data has been initiated, though it is not necessarily the
+        # case that it will be fully replicated on the filestore when queried.
+        UPLOADING = 1
+        # The data is being preprocessed.
+        PREPROCESSING = 2
+        # Validation is in progress.
+        VALIDATING = 3
+        # Import onto the production server is in progress.
+        IMPORTING = 4
+        # Postprocessing is being performed on a finished import.
+        POSTPROCESSING = 5
+        # Emails are being sent out to the relevant imported parties.
+        NOTIFYING = 6
+        # The import was successfully completed.
+        COMPLETED = 7
```

Comment:
No - the idea is that the `Status` is used to figure out if we're still working or if we have a success/failure final result,  and this can be used as a measure of "how far we got".